exposure-notification/src/main/java/fi/thl/covid19/exposurenotification/eu/FederationGatewayService.java

```diff
@@ -0,0 +1,114 @@
+package fi.thl.covid19.exposurenotification.eu;
+
+import com.google.protobuf.ByteString;
+import com.google.protobuf.InvalidProtocolBufferException;
+import fi.thl.covid19.exposurenotification.batch.BatchFileService;
+import fi.thl.covid19.exposurenotification.diagnosiskey.DiagnosisKeyDao;
+import fi.thl.covid19.exposurenotification.diagnosiskey.IntervalNumber;
+import fi.thl.covid19.exposurenotification.diagnosiskey.v1.TemporaryExposureKey;
+import fi.thl.covid19.exposurenotification.error.EfgsOperationException;
+import fi.thl.covid19.proto.EfgsProto;
+import org.springframework.stereotype.Service;
+
+import java.time.Instant;
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Optional;
+import java.util.stream.Collectors;
+
+@Service
+public class FederationGatewayService {
+
+    private final FederationGatewayClient client;
+    private final BatchFileService batchFileService;
+    private final DiagnosisKeyDao dd;
+
+    public FederationGatewayService(
+            FederationGatewayClient client,
+            BatchFileService batchFileService,
+            DiagnosisKeyDao diagnosisKeyDao
+    ) {
+        this.client = client;
+        this.batchFileService = batchFileService;
+        this.dd = diagnosisKeyDao;
+    }
+
+    public void doOutbound() {
+        long operationId = dd.startOutboundOperation();
+        try {
+            List<TemporaryExposureKey> localKeys = dd.fetchAvailableKeysForEfgs(operationId);
+            handleOutbound(transform(localKeys), operationId);
+            dd.finishOperation(operationId);
+        } catch (Throwable t) {
+            dd.markErrorOperation(operationId);
+            throw new EfgsOperationException("Outbound operation to efgs failed.", t);
+        }
+    }
+
+    public void doInbound(Optional<String> batchTag) {
+        // TODO: what date we should use?
+        String date = getDateString(LocalDate.now());
```

Comment:
The result of this is dependent on the locale of the server. You can use LocalDate.now(ZoneOffset.UTC) to get something more predictable (the current UTC-date).

Reply:
There is todo comment, so it's now just a placeholdet for actual solution.